c/test/application_base.py

```diff
@@ -0,0 +1,206 @@
+from argparse import ArgumentParser
+import copy
+import os
+import re
+import signal
+import subprocess
+import sys
+import threading
+
+
+class TestApplicationBase(object):
+    TEST_PASSED = 0
+    TEST_FAILED = 1
+    ARGUMENT_ERROR = 2
+    EXECUTION_ERROR = 3
+    NONZERO_EXIT = 4
+
+    DEFAULT_ROOT_DIR = os.path.abspath(os.path.normpath(os.path.join(os.path.dirname(__file__), '..')))
+    DEFAULT_COMMAND = ['%(path)s', '%(polaris_api_key)s', '%(unique_id)s']
+
+    def __init__(self, application_name, root_dir=None):
+        if root_dir is None:
+            self.root_dir = self.DEFAULT_ROOT_DIR
+        else:
+            self.root_dir = root_dir
+
+        self.application_name = application_name
+
+        # Define and parse arguments.
+        self.parser = ArgumentParser(usage='%(prog)s [OPTIONS]...')
+
+        self.parser.add_argument(
+            '-p', '--path', metavar='PATH',
+            help="The path to the application to be run.")
+        self.parser.add_argument(
+            '--polaris-api-key', metavar='KEY',
+            help="The Polaris API key to be used. Ignored if the POLARIS_API_KEY environment variable is specified.")
```

Comment:
This is fine as is, but I think it's generally more common to have the CLI args overried the ENV args.

Reply:
Yeah that's what I thought too, I just remembered having the opposite conversation for something else on a different repo (mapbox token possibly?). I can change this.